packages/common/src/hooks/useHostContext.ts

```diff
@@ -1,18 +1,24 @@
 import create from 'zustand';
+import { LocalStorage } from '../localStorage';
+
 import { SupportedLocales } from '../intl/intlHelpers';
 
 type HostContext = {
   locale: SupportedLocales;
   setLocale: (locale: SupportedLocales) => void;
 };
 
-const localStorageKey = '@openmsupply-client/localisation/locale';
-
 export const useHostContext = create<HostContext>(set => ({
-  locale: (localStorage.getItem(localStorageKey) as SupportedLocales) ?? 'en',
+  locale: LocalStorage.getItem('/localisation/locale') ?? 'en',
   setLocale: locale => set(state => ({ ...state, locale })),
 }));
 
 useHostContext.subscribe(({ locale }) => {
-  localStorage.setItem(localStorageKey, locale);
+  localStorage.setItem('/localisation/locale', locale);
+});
+
+LocalStorage.addListener<SupportedLocales>((key, value) => {
+  if (key === '/localisation/locale') {
+    useHostContext.setState(state => ({ ...state, locale: value }));
+  }
```

Comment:
was wondering if this is more just a 'locale' store now rather than the 'host store' but thought to wait to see if we add anymore stuff into it

Reply:
good thing the appbarbuttoncontainerref was added then!